packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -39,6 +39,11 @@ module.exports = async ({ syncToken, reporter, pluginConfig }) => {
     let errors
     if (e.code === `ENOTFOUND`) {
       details = `You seem to be offline`
+    } else if (e.code === `SELF_SIGNED_CERT_IN_CHAIN`) {
+      reporter.panic(
+        `Your self signed certificate for SSL communication is failing.`,
```

Comment:
I think this is a more descriptive error message. They still get the real error below.

```suggestion
        `We couldn't make a secure connection to your contentful space. Please check if you have any self-signed SSL certificates installed.`,
```

Reply:
Totally agree.